cpp/src/svm/svc.cu

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+
+#include "common/device_buffer.hpp"
+#include "gram/kernelfactory.h"
+#include "kernelcache.h"
+#include "label/classlabels.h"
+#include "linalg/cublas_wrappers.h"
+#include "linalg/unary_op.h"
+#include "smosolver.h"
+#include "svc.h"
+#include "svc.hpp"
+
+namespace ML {
+namespace SVM {
+
+using namespace MLCommon;
+
+template <typename math_t>
+SVC<math_t>::SVC(cumlHandle &handle, math_t C, math_t tol,
+                 GramMatrix::KernelParams kernel_params, math_t cache_size,
+                 int max_iter, bool verbose)
+  : handle(handle),
+    C(C),
+    tol(tol),
+    kernel_params(kernel_params),
+    cache_size(cache_size),
+    max_iter(max_iter),
+    verbose(verbose) {}
+
+template <typename math_t>
+SVC<math_t>::~SVC() {
+  free_buffers();
+}
+
+template <typename math_t>
+void SVC<math_t>::free_buffers() {
+  if (dual_coefs) CUDA_CHECK(cudaFree(dual_coefs));
```

Comment:
We should be using `deviceAllocator::deallocate`, because `GetDualCoeffs` is using `deviceAllocator::allocate` to create this buffer.